pkg/trace/api/otlp.go

```diff
@@ -581,33 +582,33 @@ func (o *OTLPReceiver) convertSpan(rattr map[string]string, lib pcommon.Instrume
 		// See https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#[inlineExtension]HTTP
 		if k == "http.response.status_code" {
 			gotStatusCodeFromNewConv = true
-			setMetaOTLP(span, "http.status_code", v.AsString())
+			transform.SetMetaOTLP(span, "http.status_code", v.AsString())
 		} else if k == "http.status_code" && !gotStatusCodeFromNewConv {
-			setMetaOTLP(span, "http.status_code", v.AsString())
+			transform.SetMetaOTLP(span, "http.status_code", v.AsString())
 		}
 
 		return true
 	})
 	if _, ok := span.Meta["env"]; !ok {
 		// TODO(songy23): use AttributeDeploymentEnvironmentName once collector version upgrade is unblocked
-		if _, env := getFirstFromMap(span.Meta, "deployment.environment.name", semconv.AttributeDeploymentEnvironment); env != "" {
-			setMetaOTLP(span, "env", traceutil.NormalizeTag(env))
+		if _, env := transform.GetFirstFromMap(span.Meta, "deployment.environment.name", semconv.AttributeDeploymentEnvironment); env != "" {
+			transform.SetMetaOTLP(span, "env", traceutil.NormalizeTag(env))
 		}
 	}
 	if in.TraceState().AsRaw() != "" {
-		setMetaOTLP(span, "w3c.tracestate", in.TraceState().AsRaw())
+		transform.SetMetaOTLP(span, "w3c.tracestate", in.TraceState().AsRaw())
 	}
 	if lib.Name() != "" {
-		setMetaOTLP(span, semconv.OtelLibraryName, lib.Name())
+		transform.SetMetaOTLP(span, semconv.OtelLibraryName, lib.Name())
 	}
 	if lib.Version() != "" {
-		setMetaOTLP(span, semconv.OtelLibraryVersion, lib.Version())
+		transform.SetMetaOTLP(span, semconv.OtelLibraryVersion, lib.Version())
 	}
-	setMetaOTLP(span, semconv.OtelStatusCode, in.Status().Code().String())
+	transform.SetMetaOTLP(span, semconv.OtelStatusCode, in.Status().Code().String())
 	if msg := in.Status().Message(); msg != "" {
-		setMetaOTLP(span, semconv.OtelStatusDescription, msg)
+		transform.SetMetaOTLP(span, semconv.OtelStatusDescription, msg)
 	}
-	status2Error(in.Status(), in.Events(), span)
+	transform.Status2Error(in.Status(), in.Events(), span)
 	if span.Name == "" {
 		name := in.Name()
 		if !o.conf.OTLPReceiver.SpanNameAsResourceName {
```

Comment:
There is a bug here. If we use  both `SpanNameAsResourceName` and `SpanNameRemappings`  in the config. SpanNameRemappings is using update the span name rather than incoming span name

Reply:
I'm not sure I understand the issue/what the desired behaviour is. Are you saying that if both are set, `SpanNameRemappings` should get skipped? i.e., the code should be:

```
        if span.Name == "" {
		name := in.Name()
		if !o.conf.OTLPReceiver.SpanNameAsResourceName {
			name = traceutil.OTelSpanKindName(in.Kind())
			if lib.Name() != "" {
				name = lib.Name() + "." + name
			} else {
				name = "opentelemetry." + name
			}

			// This if statement is now nested in the above
			if v, ok := o.conf.OTLPReceiver.SpanNameRemappings[name]; ok {
				name = v
			}
		}
		span.Name = name
	}
```

Reply:
the condition should always use the incoming name first. for backwards compatibility the updated name.  

```
			if v, ok := o.conf.OTLPReceiver.SpanNameRemappings[name]; ok {
				name = v
			}
```

Please add tests as well.